cfme/common/__init__.py

```diff
@@ -658,3 +658,61 @@ class UtilizationMixin(object):
     # @cached_property
     # def utilization(self):
     #     return Utilization(self)
+
+
+class CustomButtonEventsView(View):
+    """Class represents common custom button events page in CFME UI"""
+
+    breadcrumb = BreadCrumb()
+    title = Text('//div[@id="main-content"]//h1')
+    table = Table('//*[@class="miq-data-table"]/table')
+
+    @property
+    def is_displayed(self):
+        expected_title = "{} (All Custom Button Events)".format(self.context["object"].name)
+        return (
+            self.title.text == expected_title and self.breadcrumb.active_location == expected_title
+        )
+
+
+class CustomButtonEventsMixin(object):
+    @property
+    def cb_event_count(self):
+        view = navigate_to(self, "Details")
+        ent = view.entities
+        table = ent.summary("Relationships") if hasattr(ent, "summary") else ent.relationships
+        # Some objects have filed text `Custom button events` in relationship table
+        cb_text = (
+            "Custom Button Events"
+            if "Custom Button Events" in table.fields
+            else "Custom button events"
+        )
+        return int(table.get_text_of(cb_text))
+
+    def get_button_events(self):
```

Comment:
Couple of thoughts here:

1. Is a '*_count' method really necessary when this returns a list that the caller could just use len() on?

2.  Assuming we do need a '*_count' method, let's make this mixin consistent, and have it provide two properties, `button_event_count` and `button_events`.  Given the duplicated string/attribute selection in navigation and the count method, I'd lean towards keeping the Mixin skinny, and just linking Collections/Entities to a `ButtonEvents` navigation.

3. Why would `DestinationNotFound` occur when the nav class is registered a few lines below?

4. Ultimately these two convenience methods don't really buy that much convenience in the tests. The calls to navigate to 'Details' are already being made in the test, and then the value you could pull from the view you already have. Burying navigation in these convenience methods tends to end up with unexpected view changes during tests.  Convenience in one test is a poor assumption in another.

5. This class _could_ just be empty, and a placeholder to connect things to the navigation destination.